.github/workflows/check-docs-pr-preview.yml

```diff
@@ -1,64 +1,87 @@
-# This workflow checks that PRs modifying website docs include a preview link in the PR description.
-# It enforces KubeStellar's documentation PR best practices.
-
-name: Check Docs PR Preview Link
+name: Documentation Preview Automation
 
 on:
   pull_request:
-    types: [opened, edited, synchronize]
+    types: [opened, reopened, synchronize]
     paths:
-      - 'docs/content/**'
-      - 'docs/README.md'
-      - 'docs/mkdocs.yml'
-      - 'docs/overrides/**'
-      - 'docs/scripts/**'
-      - 'docs/requirements.txt'
-      - 'docs/main.py'
-      - 'SECURITY.md'
-      - 'CODE_OF_CONDUCT.md'
-      - 'GOVERNANCE.md'
-      - 'ONBOARDING.md'
+      - 'docs/**'
+      - 'website/**'
+      - '*.md'
+      - '.github/workflows/docs-preview.yml'
 
 jobs:
-  check-preview-link:
+  build-preview:
     runs-on: ubuntu-latest
     steps:
-      - name: Check out code
-        uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683
-      - name: Check for Preview Link in PR Description (bash)
-        env:
-          GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-          PR_NUMBER: ${{ github.event.pull_request.number }}
-          REPO: ${{ github.repository }}
-          BRANCH: ${{ github.head_ref }}
+      - name: Checkout code
+        uses: actions/checkout@v3
+        with:
+          ref: ${{ github.event.pull_request.head.ref }}
+          repository: ${{ github.event.pull_request.head.repo.full_name }}
+          fetch-depth: 0
+
+      - name: Setup Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: '3.10'
+
+      - name: Install dependencies
         run: |
-          set -e
-          BODY=$(gh pr view "$PR_NUMBER" --json body -q '.body')
-          DOC_BRANCH_REGEX='^doc-'
-          OWNER=$(gh pr view "$PR_NUMBER" --json headRepositoryOwner -q '.headRepositoryOwner.login')
-          REPO_NAME=$(gh pr view "$PR_NUMBER" --json headRepository -q '.headRepository.name')
-          GITHUB_PAGES_URL="https://${OWNER}.github.io/${REPO_NAME}/${BRANCH}"
-          GREEN=$'\033[0;32m'
-          RED=$'\033[0;31m'
-          YELLOW=$'\033[1;33m'
-          NC=$'\033[0m' # No Color
+          pip install -r docs/requirements.txt
+          cd website && npm install
+
+      - name: Build documentation
+        run: |
+          cd website
+          npm run build
+
+      - name: Deploy to Preview Environment
+        id: deploy
+        uses: JamesIves/github-pages-deploy-action@v4
+        with:
+          branch: gh-pages
+          folder: website/build
+          target-folder: preview/pr-${{ github.event.pull_request.number }}
+          clean: false
+          single-commit: true
```

Comment:
Why use different tools than documented in https://docs.kubestellar.io/unreleased-development/contribution-guidelines/operations/document-management/ ?